transcom/mymove
# INT B-18915 create customer form for office users

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-18915)

## Summary

In order for a SC to create a customer, they'll need a form that can submit the values that will call the API. This PR does the following:
- renames `CustomerName` component to `CreateCustomerForm` and expands to include all necessary fields to create a user and a service member
- updates routing to reflect changes

### How to test

1. Access MM and log in as an service counselor
2. Go and search for a user using an empty value
3. You should see the "Add Customer" button at the bottom of the screen
4. Click it
5. The form should appear
6. Should have all necessary validations and will be unable to submit if required values are not present

I'd make sure to `console.log(values)` in the `onSubmit` function to make sure you're getting all of the values when hitting `Save` and are getting an object with all values like so:
```
{
    "affiliation": "ARMY",
    "edipi": "1234567890",
    "first_name": "Billy",
    "middle_name": "Bobby",
    "last_name": "Brown",
    "suffix": "Mr.",
    "telephone": "999-999-9999",
    "secondary_telephone": "888-888-8888",
    "personal_email": "billybobbybrown@gmail.com",
    "phone_is_preferred": true,
    "email_is_preferred": true,
    "residential_address": {
        "streetAddress1": "1234 Funky St.",
        "streetAddress2": "Apartment Boogaloo",
        "streetAddress3": "C/O Funkmaster",
        "city": "Funky",
        "state": "LA",
        "postalCode": "70726"
    },
    "backup_mailing_address": {
        "streetAddress1": "4567 Groovy Ave.",
        "streetAddress2": "Apt. Funky Four",
        "streetAddress3": "C/O Groovemaster",
        "city": "Groovy",
        "state": "AK",
        "postalCode": "99540"
    },
    "backup_contact": {
        "name": "Backup Dancers",
        "telephone": "777-777-7777",
        "email": "dancewithmenow@gmail.com"
    },
    "create_okta_account": "true"
}
```

## Screenshots

https://github.com/transcom/mymove/assets/136510600/eb0e340c-7ac3-4b26-8d1d-7087534a54da
